### PR TITLE
Cogs and extension docs fixes

### PR DESCRIPTION
**Pull request by laggycomputer**
I have fixed grammar, punctuation, and capitalization (once or twice). I have tried my best to keep consistency and use `you` instead of `we`. I have also introduced slight wording changes.

Is my grammar trash? Do the docs now sound too commanding? Kindly give feedback.

### CODE REVIEW

File: docs/ext/commands/extensions.rst [hunk 1]
```diff
@@ -36,24 +36,24 @@ In this example we define a simple command, and when the extension is loaded thi
...
-The act of reloading an extension is actually quite simple -- it is as simple as unloading it and then reloading it.
+The act of reloading an extension is quite simple -- it is as simple as unloading and loading again:
 
 .. code-block:: python3
 
     >>> bot.unload_extension('hello')
     >>> bot.load_extension('hello')
 
-Once we remove and load the extension, any changes that we did will be applied upon load. This is useful if we want to add or remove functionality without restarting our bot.
+Once you remove and load the extension, any changes that made to the extension code will be applied. This is useful if you want to add or remove functionality without restarting our bot.
```
Comment (Rapptz) [inline docs/ext/commands/extensions.rst]: This is just grammatically incorrect.

File: docs/ext/commands/extensions.rst [hunk 2]
```diff
@@ -36,24 +36,24 @@ In this example we define a simple command, and when the extension is loaded thi
...
 
 Cleaning Up
 -------------
 
-Although rare, sometimes an extension needs to clean-up or know when it's being unloaded. For cases like these, there is another entry point named ``teardown`` which is similar to ``setup`` except called when the extension is unloaded.
+Although rare, sometimes an extension needs to run some clean-up code before unloading. For cases like these, there is another entry point named ``teardown`` which is similar to ``setup``, but called when the extension is *un*loaded.
```
Comment (Rapptz) [inline docs/ext/commands/extensions.rst]: This doesn't buy anything over the previous wording.

File: docs/ext/commands/extensions.rst [hunk 3]
```diff
@@ -36,24 +36,24 @@ In this example we define a simple command, and when the extension is loaded thi
...
 .. note::
 
-    Extension paths are ultimately similar to the import mechanism. What this means is that if there is a folder, then it must be dot-qualified. For example to load an extension in ``plugins/hello.py`` then we use the string ``plugins.hello``.
+    Extension paths are ultimately similar to the import mechanism. This includes dot-qualifying files in subdirectories. For example, to load an extension in ``plugins/hello.py``, we use the string ``plugins.hello``.
 
 Reloading
 -----------
 
-The act of reloading an extension is actually quite simple -- it is as simple as unloading it and then reloading it.
+The act of reloading an extension is quite simple -- it is as simple as unloading and loading again:
```
[the remark this answers is not in the document]
Comment (laggycomputer) [reply docs/ext/commands/extensions.rst]: I think this case is something like the rule to never use a word in its definition.
If you prefer the former wording, it's fine with me.

**Comment by laggycomputer** [comment]
Better...?

**Comment by laggycomputer** [comment]
It seems time to close this, as it is almost a stale as month-old bread.

**Pull request closed without merging**